internal/filter/filterottl/filter.go

```diff
@@ -129,23 +124,14 @@ func NewBoolExprForResource(conditions []string, functions map[string]ottl.Facto
 	if _, ok := functions[match.Name()]; !ok {
 		functions[match.Name()] = match
 	}
-	statmentsStr := conditionsToStatements(conditions)
 	parser, err := ottlresource.NewParser(functions, set)
 	if err != nil {
 		return nil, err
 	}
-	statements, err := parser.ParseStatements(statmentsStr)
+	statements, err := parser.ParseConditions(conditions)
 	if err != nil {
 		return nil, err
 	}
-	s := ottlresource.NewStatements(statements, set, ottlresource.WithErrorMode(errorMode))
-	return &s, nil
-}
-
-func conditionsToStatements(conditions []string) []string {
-	statements := make([]string, len(conditions))
-	for i, condition := range conditions {
-		statements[i] = "match() where " + condition
```

Comment:
Are the non-OTTL changes just illustrative, or should we also remove the `match` definition in this PR?

Reply:
Good catch, I can remove that too